### PR TITLE
Hb 04

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
Great job!

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("from User u where "
+                    + "u.email = :email");
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            Query<User> query = session.createQuery("from User u where "
                    + "u.email = :email", User.class);
```

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            query.setParameter("email", email);
+            User user = (User) Optional.of(query.getSingleResult()).get();
+            return Optional.ofNullable(user);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            return query.uniqueResultOptional();
```

File: src/main/java/com/dev/cinema/model/User.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "user")
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/User.java]: remove empty line

File: src/main/java/com/dev/cinema/model/User.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(unique = true,name = "email")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/User.java]:
```suggestion
    @Column(unique = true, name = "email")
```

File: src/main/java/com/dev/cinema/util/HashUtil.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        StringBuilder hashPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(HASHING_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Hashing password was failed.");
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/util/HashUtil.java]:
```suggestion
            throw new RuntimeException("Hashing password was failed.", e);
```

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -39,11 +39,10 @@ public User add(User user) {
...
     public Optional<User> findByEmail(String email) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            Query query = session.createQuery("from User u where "
-                    + "u.email = :email");
+            Query<User> query = session.createQuery("from User u where "
+                    + "u.email = :email", User.class);
             query.setParameter("email", email);
-            User user = (User) Optional.of(query.getSingleResult()).get();
-            return Optional.ofNullable(user);
+            return Optional.ofNullable(query.getSingleResult());
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: ??? https://github.com/IrynaDim/Movie-Shop/pull/4#discussion_r501494532

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**